src/platform/cc13x2_26x2/BLEManagerImpl.cpp

```diff
@@ -1803,6 +1812,55 @@ void BLEManagerImpl::CHIPoBLEProfile_charValueChangeCB(uint8_t paramId, uint16_t
     }
 }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+/*********************************************************************
+ * @fn      CHIPoBLEProfile_c3ValueReadCB
+ *
+ * @brief   Callback from CHIPoBLE Profile indicating a commissioner
+ *          reading an additionald commissioning-related data (C3) from
+ *          commissionee.
+ *          Calling context (BLE Stack Task)
+ *
+ * @param   destBuf - buffer to copy data to
+ *          maxLen - maximum length of destBuf
+ *
+ * @return  None.
+ */
+void BLEManagerImpl::CHIPoBLEProfile_c3ValueReadCB(uint8_t * destBuf, uint16_t maxLen, uint16_t connHandle)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    System::PacketBufferHandle packetBuf;
+    uint16_t len;
```

Comment:
declare this where used not on top of the file. that reduces scope and makes this feel more C++ than C.

Reply:
I see that you do this because `goto exit`. Could we not use goto-exit?

Reply:
> declare this where used not on top of the function. that reduces scope and makes this feel more C++ than C.

Ok, we can move the variables to feel more like C++.
Specifically:
- we can move declaration of `packetBuf` to just before the `generateAdditionalDataPayload`
- we can move declaration of `len` to just before the `len =`

> I see that you do this because `goto exit`. Could we not use goto-exit?

What do you mean by `goto-exit`? How is this different from `goto exit` ?
My initial guess is that you mean to replace `SuccessOrExit(err);` with `goto exit` -- this way the reference to `err` is not needed when resolving macros.
So, I will do this for now, and if you mean differently, then I can re-consider it!

Reply:
Partially handled here: https://github.com/ti-tpan/connectedhomeip/commit/472f439e3ae0c0da2c18f4c59a1024af3d2d3daa